src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,176 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int START_CAPACITY = 16;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = new Node[START_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            int index = 0;
+            if (table[index] == null) {
+                table[index] = new Node<>(null, value);
+            } else {
+                Node<K, V> currentNode = table[index];
+                while (currentNode != null) {
+                    if (currentNode.key == null) {
+                        currentNode.value = value;
+                        return;
+                    }
+                    currentNode = currentNode.next;
+                }
+                Node<K, V> newNode = new Node<>(null, value);
+                newNode.next = table[index];
+                table[index] = newNode;
+            }
+            size++;
+            if ((double) size / table.length > 0.75) {
+                resize();
+            }
+            return;
+        }
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if (currentNode.key != null && currentNode.key.equals(key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+            currentNode = table[index];
+            while (currentNode.next != null) {
+                currentNode = currentNode.next;
+            }
+            currentNode.next = newNode;
+        }
+        size++;
+        if ((double) size / table.length > 0.75) {
+            resize();
```

Comment:
too much code, also its duplicated, you can make this method fit in less than 20 lines